lms/djangoapps/courseware/entrance_exams.py

```diff
@@ -0,0 +1,173 @@
+    """
+    exam_module_ids = [exam_module.location for exam_module in exam_modules]
+    student_modules = StudentModule.objects.filter(
+        student=user,
+        course_id=course_descriptor.id,
+        module_state_key__in=exam_module_ids,
+    )
+    exam_pct = 0
+    if student_modules:
+        module_pcts = []
+        ignore_categories = ['course', 'chapter', 'sequential', 'vertical']
+        for module in exam_modules:
+            if module.graded and module.category not in ignore_categories:
+                module_pct = 0
+                try:
+                    student_module = student_modules.get(module_state_key=module.location)
+                    if student_module.max_grade:
+                        module_pct = student_module.grade / student_module.max_grade
+                    module_pcts.append(module_pct)
+                except StudentModule.DoesNotExist:
+                    pass
+        if module_pcts:
+            exam_pct = sum(module_pcts) / float(len(module_pcts))
+    return exam_pct
+
+
+def get_entrance_exam_score(request, course):
+    """
+    Get entrance exam score
+    """
+    from courseware.module_render import get_module_for_descriptor
+    exam_key = UsageKey.from_string(course.entrance_exam_id)
+    exam_descriptor = modulestore().get_item(exam_key)
+
+    def inner_get_module(descriptor):
+        """
+        Delegate to get_module_for_descriptor.
+        """
+        field_data_cache = FieldDataCache([descriptor], course.id, request.user)
+        return get_module_for_descriptor(request.user, request, descriptor, field_data_cache, course.id)
+
+    exam_module_generators = yield_dynamic_descriptor_descendents(
+        exam_descriptor,
+        inner_get_module
+    )
+    exam_modules = [module for module in exam_module_generators]
+    return calculate_entrance_exam_score(request.user, course, exam_modules)
+
+
+def get_entrance_exam_content_info(request, course):
+    """
+    Get the entrance exam content information e.g. chapter, exam passing state.
+    return exam chapter and its passing state.
+    """
+    required_content = get_required_content(course, request.user)
+    exam_chapter = None
+    is_exam_passed = True
+    # Iterating the list of required content of this course.
+    for content in required_content:
+        # database lookup to required content pointer
+        usage_key = course.id.make_usage_key_from_deprecated_string(content)
+        module_item = modulestore().get_item(usage_key)
+        if not module_item.hide_from_toc and module_item.is_entrance_exam:
+            # Here we are looking for entrance exam module/chapter in required_content.
+            # If module_item is an entrance exam chapter then set and return its info e.g. exam chapter, exam state.
+            exam_chapter = module_item
+            is_exam_passed = False
+            break
+    return exam_chapter, is_exam_passed
+
+
+def get_required_content(course, user):
```

Comment:
Okay, I think maybe I was a little unclear on how much to move from milestones_helpers.py into this new module -- direct interactions with the milestones API should be contained within milestones_helpers.py.  Logic specific to entrance exams which build on top of that logic should make calls to milestones_helpers (eg, get_required_content) and then perform any necessary post-processing.  Perhaps a good way to think about this is:  If we are checking for MILESTONES_APP, it should be in milestones_helpers.py.  If we are checking for ENTRANCE_EXAMS, it should be in entrance_exams.py.